File: lib/playqueensgame_site_kit.ex
defmodule PlayQueensGameSiteKit do
  @moduledoc """
  Small URL helpers for [Queens Game](https://playqueensgame.org).
  """

  @base_url "https://playqueensgame.org"

  def home_url, do: @base_url

  def page_url(slug) when is_binary(slug) do
    clean = String.trim(slug, "/")
    if clean == "", do: @base_url, else: @base_url <> "/" <> clean <> "/"
  end

  def play_url, do: @base_url <> "/#play"

  def daily_url, do: page_url("daily")

  def learn_url, do: page_url("learn")

  def blog_url, do: page_url("blog")
end

File: lib/playqueensgame_site_kit.dart
const baseUrl = 'https://playqueensgame.org';

String homeUrl() => baseUrl;

String pageUrl(String slug) {
  final clean = slug.replaceAll(RegExp(r'^/+|/+$'), '');
  return clean.isEmpty ? baseUrl : '$baseUrl/$clean/';
}

String playUrl() => '$baseUrl/#play';

String dailyUrl() => pageUrl('daily');

String learnUrl() => pageUrl('learn');

String blogUrl() => pageUrl('blog');
